lib/internal/url.js

```diff
@@ -1414,25 +1414,27 @@ const backslashRegEx = /\\/g;
 const newlineRegEx = /\n/g;
 const carriageReturnRegEx = /\r/g;
 const tabRegEx = /\t/g;
+const questionRegex = /\?/g;
+const hashRegex = /#/g;
 
 function encodePathChars(filepath) {
-  if (StringPrototypeIncludes(filepath, '%'))
+  if (StringPrototypeIndexOf(filepath, '%') !== -1)
     filepath = RegExpPrototypeSymbolReplace(percentRegEx, filepath, '%25');
   // In posix, backslash is a valid character in paths:
-  if (!isWindows && StringPrototypeIncludes(filepath, '\\'))
+  if (!isWindows && StringPrototypeIndexOf(filepath, '\\') !== -1)
     filepath = RegExpPrototypeSymbolReplace(backslashRegEx, filepath, '%5C');
-  if (StringPrototypeIncludes(filepath, '\n'))
+  if (StringPrototypeIndexOf(filepath, '\n') !== -1)
     filepath = RegExpPrototypeSymbolReplace(newlineRegEx, filepath, '%0A');
```

Comment:
Just for my curiosity: Isnt it faster to use String.replaceAll ?

Reply:
Benchmarks do not show any noticable difference, but I'll follow up on a different pull request and run the benchmark CI on this.